x-pack/plugins/alerts/common/alert_notify_when_type.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+const AlertNotifyWhenTypeValues = ['onActionGroupChange', 'onActiveAlert', 'onThrottleInterval'];
+export type AlertNotifyWhenType = typeof AlertNotifyWhenTypeValues[number];
```

Comment:
I think this typing is wrong.
It makes any `string` a valid value for notifyWhen.

I think you meant:
```suggestion
const AlertNotifyWhenTypeValues = [
  'onActionGroupChange',
  'onActiveAlert',
  'onThrottleInterval',
] as const;
export type AlertNotifyWhenType = typeof AlertNotifyWhenTypeValues[number];
```

Reply:
Good catch! Updated.